### ISSUE TITLE
Use helm chart for deploying garm-operator

### ISSUE THREAD

**Issue by gustaff-weldon** [mercedes-benz/garm-operator#138]
### What is the feature you would like to have?

Instead of deploying manifests which first have to be processed via `envsubst` garm-operator could be released as helm chart where all necessary customisations can be done via values.

**Comment by bavarianbidi** [comment]
the operator got created with `kubebuilder`.  And `kubebuilder` is taking care of updating manifests when certain changes in the code happens.

The biggest concern about having a helm-chart as well is the fact, that a helm-chart needs manual maintenance.
We've initially take a look at [helmify](https://github.com/arttor/helmify) to generate a helm-chart as well but decided to not use it for now. But this is more a reason from a capacity point of view rather than a "we do not want to have this" reason :sweat_smile: 

FYI: There is currently an ongoing discussion to generate `helm` charts with `kubebuilder` (xref: [✨[WIP]feat: Add the proposal about the helm chart autogenerate plugin.](https://github.com/kubernetes-sigs/kubebuilder/pull/3632#top)). Once this has landed in `kubebuilder`, i'm sure we will adapt this feature as well.

cc @H777K as my favorite `helm` user ;-)

**Comment by gabriel-samfira** [comment]
I _think_ this landed in the master branch: https://github.com/kubernetes-sigs/kubebuilder/tree/master/pkg/plugins/optional/helm

Can hardly wait for the next stable release

**Comment by bavarianbidi** [comment]
yep, i've recently gave them a try but there is still some work to do in kubebuilder to get this working without refactoring stuff in an existing repo.
And kubebuilder has to be build manually as it's not released yet ( my try was more than a month ago)

**Comment by gabriel-samfira** [comment]
That's still progress, which is awesome!

**Comment by chickenkiller** [comment]
I could generate a helm chart using the latest kubebuilder, and following their instructions which are
```sh
make manifests
kubebuilder edit --plugins=helm/v1-alpha
```
But I definitely don't have the knowledge to check if the resulting helm chart is proper or if it will behave correctly. But I think it could be a good time for garm-operator maintainers to have another try at this helm plugin 😉

**Comment by bavarianbidi** [comment]
waiting for https://github.com/kubernetes-sigs/kubebuilder/pull/4891 to get merged

**Comment by bavarianbidi** [comment]
As the `helm/v1-alpha` plugin recently got re-written and we now have `helm/v2-alpha` it should be an easy one to create valid helm-charts from the existing manifests.

unfortunately, as still a kubebuilder-release is missing, we can build `kubebuilder` by our own and generate the helm-chart once.

xref:
* https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/plugins/available/helm-v2-alpha.md
